packages/cli/src/lib/setup/setupInstall.ts

```diff
@@ -385,17 +424,67 @@ export class Install {
                 console.error(`host.${hostname} Cannot install ${npmUrl}: ${result.exitCode}`);
                 return this.processExit(EXIT_CODES.CANNOT_INSTALL_NPM_PACKET);
             }
-            // create file that indicates, that npm was called there
+            // create file that indicates that npm was called there
             fs.writeFileSync(path.join(installDir, 'iob_npm.done'), ' ');
             // command succeeded
             return { _url: npmUrl, installDir: path.dirname(installDir) };
         } else {
+            const adapterName = this.getAdapterNameFromUrl(npmUrl);
+            if (adapterName && !isRetry && result.stderr.includes('ENOTEMPTY')) {
```

Comment:
Do we need the adaptername with the new aproach?

Reply:
No, will change